### PR TITLE
[MM-65645] Remove unnecessary URL validation for external links, focus only on where security issues exist and allow special characters to be handled as needed

### PR DESCRIPTION
**Pull request by devinbinnie**
#### Summary
The initial fix for MM-65645 ([#3685](https://github.com/mattermost/desktop/pull/3685)) addressed Microsoft product links by normalizing specific characters before strict RFC 3986 validation, but this approach required adding new workarounds for each non-standard URI pattern.

This PR replaces that approach by not enforcing the `isValidURI` check for external links, since it doesn't directly present a security issue. Instead, some additional changes were made that addresses where security issues could crop up while also allowing the URLs to work correctly:
- Fixed the `parseURL` duplicate-slash regex to preserve `scheme:///` patterns
- Always passes the serialized URL to `shell.openExternal` (percent-encoding dangerous characters like backticks, spaces, and quotes)
- Rejects URLs with null bytes

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-65645
Closes https://github.com/mattermost/desktop/issues/3525

```release-note
Fixed further issues around URL parsing with special characters
```

## Change Impact: 🔴 High

**Reasoning:** The changes modify a shared URL utility (parseURL), remove normalization logic used across the codebase, and change a public API signature (allowProtocolDialog.handleDialogEvent), impacting navigation, protocol handling, and external-open flows across multiple modules and processes.

**Regression Risk:** High. Widespread parsing and external-opening paths are altered (including handling of special characters and null bytes), there's a breaking API change, and edge cases—especially interactions with native apps and non-standard URIs—may be insufficiently covered by tests.

** QA Recommendation:** Require extensive manual QA plus expanded automated tests. Prioritize validation of standard and non-standard protocols (http/https, mailto, onenote, custom), navigation/new-window/popup flows, shell.openExternal behavior with special and non-ASCII characters, null-byte rejection, and compatibility of protocol-dialog integrations. Do not skip manual QA.

*Generated by CodeRabbitAI*

### CODE REVIEW

File: src/app/views/pluginsPopUps.ts
Comment (edgarbellot) [inline src/app/views/pluginsPopUps.ts]: @devinbinnie Should we do `shell.openExternal(parsedURL.toString());` here to make sure we are using the serialized form and not the raw URL?

Comment (devinbinnie) [reply src/app/views/pluginsPopUps.ts]: Probably not a concern since it should be caught by the other checks first, but definitely won't hurt to do so so let's do it.
```suggestion
                shell.openExternal(parsedURL.toString());
```

File: src/app/views/pluginsPopUps.ts [hunk 1]
```diff
@@ -93,7 +93,7 @@ export class PluginsPopUpsManager {
                 NavigationManager.openLinkInNewTab(parsedURL);
             } else {
                 // We allow to open external links through browser.
-                shell.openExternal(url);
+                shell.openExternal(parsedURL.toString());
```
[the remark this answers is not in the document]
Comment (devinbinnie) [reply src/app/views/pluginsPopUps.ts:96]: @coderabbit This is not necessary. Resolve.

**Review by wiggin77 — APPROVED** [review]
